### DISCUSSION TITLE
Tensor dimension mismatch error, Expected size for first two dimensions of batch2 tensor error and so.

### DISCUSSION THREAD

**Discussion by ahmed-ansari-31** [openai/whisper#1392] [Q&A]
I am using flask_socketio for sending audio: 
Python == 3.10.7 + Flask 

1. Whisper throws tensor dimension mis-match while processing longer audios ?
2. Also Whisper throws tensor dimension mis-match while processing longer audios ?
3. KeyError: Linear(in_features=768, out_features=768, bias=True)
4. RuntimeError: Expected size for first two dimensions of batch2 tensor to be: [12, 9] but got: [12, 17]. # Some times these numbers change.

**Comment by claudinestennett1998** [comment]
The "tensor size mismatch" error occurs when the dimensions of tensors do not match when performing operations or passing data. This can happen, for example, when you try to perform operations on tensors of different sizes or pass tensors with the wrong dimensions to a machine learning model. Personally, I have encountered this problem, and it requires care when working with tensors.

By the way, I found here a great photo of BMW in vector format. It inspired me to develop a video game dedicated to BMW, in which you can enjoy racing, tuning and car maintenance.

**Reply by ahmed-ansari-31** [reply]
you have encountered this problem, So how did you overcome with it?
And from where THE BMW came from in this discussion?

**Reply by phineas-pta** [reply]
the comment is obviously AI generated

**Comment by pierrot-le-foo** [comment]
Not sure if that helps but maybe you can increase sample length:

```
model.transcribe(sound_file, sample_len=500)
```
